src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```
@@ -360,6 +358,58 @@ private static void AddProjectFileDependenciesForPackageReference(PackageSpec pr
             }
         }
 
+        private void AddCentralTransitiveDependencyGroupsForPackageReference(PackageSpec project, LockFile lockFile, IEnumerable<RestoreTargetGraph> targetGraphs)
+        {
+            if(project.RestoreMetadata?.CentralPackageVersionsEnabled == false)
+            {
+                return;
+            }
+
+            // Do not pack anything from the runtime graphs
+            // The runtime graphs are added in addition to the graphs without a runtime 
+            foreach (var targetGraph in targetGraphs
+                .Where(targetGraph => string.IsNullOrEmpty(targetGraph.RuntimeIdentifier))
+                .OrderBy(graph => graph.Framework.ToString(), StringComparer.Ordinal))
+            {
+                if (_includeFlagGraphs.TryGetValue(targetGraph, out Dictionary<string, LibraryIncludeFlags> dependenciesIncludeFlags))
```

Similar feedback as before, rely on the value being available. 
We'd rather throw and fail fast than silently give a wrong result. 
This should never be the case :)

---

At the minimum, ignoring a package if we can't find the flag seems like the wrong thing.

---

Agree - removed.